src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -127,28 +222,28 @@ void SilabsTracer::TimeTraceEnd(TimeTraceOperation aOperation, CHIP_ERROR error)
         }
     }
 
-    OutputTrace(tracker);
+    return OutputTrace(tracker);
 }
 
-void SilabsTracer::TimeTraceInstant(TimeTraceOperation aOperation, CHIP_ERROR error)
+CHIP_ERROR SilabsTracer::TimeTraceInstant(TimeTraceOperation aOperation, CHIP_ERROR error)
 {
     TimeTracker tracker;
     tracker.mStartTime = System::SystemClock().GetMonotonicTimestamp();
     tracker.mEndTime   = tracker.mStartTime;
     tracker.mOperation = aOperation;
     tracker.mType      = OperationType::kInstant;
     tracker.mError     = error;
-    OutputTrace(tracker);
+    return OutputTrace(tracker);
 }
 
 CHIP_ERROR SilabsTracer::OutputTimeTracker(const TimeTracker & tracker)
 {
     VerifyOrReturnError(isLogInitialized(), CHIP_ERROR_UNINITIALIZED);
-    ChipLogProgress(DeviceLayer,
-                    "TimeTracker - StartTime: %" PRIu32 ", EndTime: %" PRIu32 ", Operation: %" PRIu8 ", Type: %" PRIu8
-                    ", Error: %" CHIP_ERROR_FORMAT,
-                    tracker.mStartTime.count(), tracker.mEndTime.count(), static_cast<uint8_t>(tracker.mOperation),
-                    static_cast<uint8_t>(tracker.mType), tracker.mError.Format());
+    // Allocate a buffer to store the trace
+    char buffer[256];
```

Comment:
would it be useful to have a constant for the buffer size? passing a bytespan and using bytespans in the function would make sense to me